includes/core/classes/class-event-feed.php

```diff
@@ -0,0 +1,259 @@
+<?php
+/**
+ * Handles event feeds for GatherPress.
+ *
+ * This class is responsible for improving the default RSS feeds for events,
+ * including filtering to show only upcoming events, ordering by start date,
+ * and customizing excerpts to show event details.
+ *
+ * @package GatherPress\Core
+ * @since 1.0.0
+ */
+
+namespace GatherPress\Core;
+
+// Exit if accessed directly.
+defined( 'ABSPATH' ) || exit; // @codeCoverageIgnore
+
+use GatherPress\Core\Traits\Singleton;
+use WP_Query;
+
+/**
+ * Class Event_Feed.
+ *
+ * Manages event feeds for GatherPress.
+ *
+ * @since 1.0.0
+ */
+class Event_Feed {
+	/**
+	 * Enforces a single instance of this class.
+	 */
+	use Singleton;
+
+	/**
+	 * Class constructor.
+	 *
+	 * This method initializes the object and sets up necessary hooks.
+	 *
+	 * @since 1.0.0
+	 */
+	protected function __construct() {
+		$this->setup_hooks();
+	}
+
+	/**
+	 * Set up hooks for various purposes.
+	 *
+	 * This method adds hooks for different purposes as needed.
+	 *
+	 * @since 1.0.0
+	 *
+	 * @return void
+	 */
+	protected function setup_hooks(): void {
+		// Customize RSS excerpts for events.
+		add_filter( 'the_excerpt_rss', array( $this, 'customize_event_excerpt' ) );
+		add_filter( 'the_content_feed', array( $this, 'customize_event_content' ) );
+
+		// Hook into the main query to handle events feeds.
+		add_action( 'pre_get_posts', array( $this, 'handle_events_feed_query' ) );
+
+		// Add custom rewrite rules for events feed.
+		add_action( 'init', array( $this, 'add_events_feed_rewrite_rules' ) );
+	}
+
+	/**
+	 * Handle events feed queries by setting the appropriate parameters.
+	 *
+	 * This method works in conjunction with Event_Query::prepare_event_query_before_execution()
+	 * to ensure feeds show upcoming events with proper sorting.
+	 *
+	 * @since 1.0.0
+	 *
+	 * @param WP_Query $query The WP_Query instance.
+	 * @return void
+	 */
+	public function handle_events_feed_query( WP_Query $query ): void {
+		// Only run on the main query and if it's a feed.
+		if ( ! $query->is_main_query() || ! $query->is_feed ) {
+			return;
+		}
+
+		// Check if this is an events feed request.
+		if ( isset( $_SERVER['REQUEST_URI'] ) ) {
+			$request_uri = sanitize_text_field( wp_unslash( $_SERVER['REQUEST_URI'] ) );
+
+			// Check if this is /events/feed/.
+			if ( strpos( $request_uri, '/events/feed' ) !== false ) {
```

Comment:
It might be better to build the URL based on the gatherpress_general['urls']['events'] setting and not with the hardcoded 'events' string.

Reply:
@carstingaxion I am using the setting now.